packages/arb-token-bridge-ui/src/token-bridge-sdk/v2/BridgeTransferStarterFactoryV2.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  BridgeTransferStarterV2,
+  BridgeTransferStarterV2Props
+} from './BridgeTransferStarterV2'
+import {
+  isTokenArbitrumGoerliNativeUSDC,
+  isTokenArbitrumOneNativeUSDC,
+  isTokenGoerliUSDC,
+  isTokenMainnetUSDC
+} from '../../util/TokenUtils'
+import { isNetwork } from '../../util/networks'
+import { EthDepositStarterV2 } from './EthDepositV2'
+import { Erc20DepositStarterV2 } from './Erc20DepositV2'
+import { TokenType } from '../../hooks/arbTokenBridge.types'
+import { CctpDepositStarterV2 } from './CctpDepositV2'
+import { CctpWithdrawalStarterV2 } from './CctpWithdrawalV2'
+import { EthWithdrawalStarterV2 } from './EthWithdrawalV2'
+import { Erc20WithdrawalStarterV2 } from './Erc20WithdrawalV2'
+
+export class BridgeTransferStarterFactoryV2 {
+  public static async init(
+    props: BridgeTransferStarterV2Props
+  ): Promise<BridgeTransferStarterV2 | undefined> {
+    const { sourceChainProvider, destinationChainProvider, selectedToken } =
+      props
+
+    const sourceChainNetwork = await sourceChainProvider.getNetwork()
+    const destinationChainNetwork = await destinationChainProvider.getNetwork()
+
+    const sourceChainId = sourceChainNetwork.chainId
+    const destinationChainId = destinationChainNetwork.chainId
+
+    const isBaseChainEthereum =
+      isNetwork(sourceChainId).isEthereumMainnetOrTestnet
+    const isBaseChainArbitrum = isNetwork(sourceChainId).isArbitrum
```

Comment:
i think Base Chain means Ethereum chains?
we probably want to call this "Source chain"